src/domain/johtoselvitys/JohtoselvitysForm.test.tsx

```diff
@@ -132,3 +134,48 @@ test('Cable report application form can be filled and saved and sent to Allu', a
   await user.click(screen.getByRole('button', { name: /lähetä hakemus/i }));
   expect(screen.queryByText(/Hakemus lähetetty/i)).toBeInTheDocument();
 });
+
+test('Should show error message when saving fails', async () => {
+  const user = userEvent.setup();
+
+  server.use(
+    rest.post('/api/hakemukset', async (req, res, ctx) => {
+      return res(ctx.status(500), ctx.json({ errorMessage: 'Failed for testing purposes' }));
+    })
+  );
+
+  render(<Johtoselvitys />, undefined, '/fi/johtoselvityshakemus?hanke=HAI22-2');
+
+  // Fill basic information page, so that form can be saved for the first time
+  fireEvent.change(screen.getByLabelText(/työn nimi/i), {
```

Comment:
Sonar complains about rows 150 - 178 to be duplicated. Could this part be extracted to a helper method etc?

Reply:
Extracted that to a helper method.